smartpower3/wifimanager.h

```diff
@@ -47,54 +33,56 @@ const char WIFI_CMD_MENU[][50] = {
     "7. WiFi Command mode exit"
 };
 
-#define WIFI_CMD_MENU_CNT   (sizeof(WIFI_CMD_MENU)/sizeof(WIFI_CMD_MENU[0]))
-
-
 class WiFiManager
 {
 public:
 	WiFiManager(WiFiUDP &udp, WiFiClient &client);
 	void view_main_menu(void);
-	void view_ap_list(int ap_list_cnt);
-	void ap_scanning(void);
-	bool ap_connect(int ap_number, char *passwd);
-	bool ap_connect(String ssid, String passwd);
-	void ap_set_passwd(int ap_number);
-	void ap_select(int ap_list_cnt);
+	void view_ap_list(int16_t ap_list_cnt);
+	int16_t ap_scanning(void);
+	bool ap_connect(uint8_t ap_number, char *passwd);
+	bool ap_connect(String ssid, String passwd, bool show_ctrl_c_info = false);
+	void ap_set_passwd(uint8_t ap_number);
+	void ap_select(int16_t ap_list_cnt);
+	void ap_select_and_connect(void);
 	void ap_forget(void);
-	void ap_info(int ap_number);
+	void ap_info(void);
 	void udp_server_info();
 	void udp_server_forget(void);
 	void cmd_main(char idata);
 	bool isCommandMode(void);
 	void setCommandMode(void);
 	void set_udp();
-	uint8_t state = 0;
-	uint8_t nvs_state = 0;
-	//void setConnType(CONN_TYPE conn_type);
+	wifi_credentials_state credentials_state = STATE_CREDENTIALS_OK;
 	uint16_t port_udp = 0;
 	IPAddress ipaddr_udp;
 	bool update_udp_info = true;
 private:
 	WiFiUDP udp;
 	WiFiClient client;
-	uint16_t WiFiState = eSTATE_WIFI_IDEL;
-
-	int ConnectAP_Number = 0, ConnectAP_RSSI = 0, APListCount = 0, TCPTestCount = 0;
-	char ConnectAP_Passwd[64] = {0,};
-	bool isConnectedAP = false, isTCPTest = false;
```

Comment:
most were not used or used only locally (so no need to save them as object properties)